tasks/localizationTasks.ts

```diff
@@ -0,0 +1,174 @@
+        for (const file of locFiles) {
+            files.push(util.format(file, lang));
+        }
+    }
+    // English
+    files.push('bundle.l10n.json');
+    return files;
+}
+
+async function git_diff(args: string[]): Promise<string[]> {
+    const result = await git(['diff'].concat(args));
+    // Line ending from the stdout of git is '\n' even on Windows.
+    return result
+        .replaceAll('\n', EOL)
+        .split(EOL)
+        .map((fileName, _) => fileName.trim())
+        .filter((fileName) => fileName.length !== 0);
+}
+
+async function git(args: string[], printCommand = true): Promise<string> {
+    if (printCommand) {
+        console.log(`git ${args.join(' ')}`);
+    }
+
+    const git = spawnSync('git', args);
+    if (git.status != 0) {
+        const err = git.stderr.toString();
+        if (printCommand) {
+            console.log(`Failed to execute git ${args.join(' ')}.`);
+        }
+        throw err;
+    }
+
+    const stdout = git.stdout.toString();
+    if (printCommand) {
+        console.log(stdout);
+    }
+    return stdout;
+}
+
+gulp.task('publish localization content', async () => {
+    const parsedArgs = minimist<Options>(process.argv.slice(2));
+    await git(['add', '-A']);
+    const diffResults = await git_diff(['--name-only', 'HEAD']);
+    if (diffResults.length == 0) {
+        console.log('No git file changes');
+        return;
+    }
+
+    const localizationChanges = getGeneratedLocalizationChanges(diffResults);
+    if (localizationChanges.length == 0) {
+        console.log('No localization file changed');
+        return;
+    }
+    await git(['reset']);
+    const newBranchName = `localization/${parsedArgs.commitSha}`;
+    await git(['config', '--local', 'user.name', parsedArgs.userName]);
+    await git(['config', '--local', 'user.email', parsedArgs.email]);
+
+    console.log(`Changed files going to be staged: ${localizationChanges}`);
+    await git(['add'].concat(localizationChanges));
+    await git(['checkout', '-b', newBranchName]);
+    await git(['commit', '-m', `Localization result of ${parsedArgs.commitSha}.`]);
+
+    const pat = process.env['GitHubPAT'];
+    if (!pat) {
+        throw 'No GitHub Pat found.';
+    }
+
+    const remoteRepoAlias = 'targetRepo';
+    await git(
+        [
+            'remote',
+            'add',
+            remoteRepoAlias,
+            `https://${parsedArgs.userName}:${pat}@github.com/${parsedArgs.owner}/${parsedArgs.targetRemoteRepo}.git`,
+        ],
+        // Note: don't print PAT to console
+        false
+    );
+    await git(['fetch', remoteRepoAlias]);
+
+    const lsRemote = await git(['ls-remote', remoteRepoAlias, 'refs/head/' + newBranchName]);
+    if (lsRemote.trim() !== '') {
+        // If the localization branch of this commit already exists, don't try to create another one.
+        console.log(`${newBranchName} already exists in ${parsedArgs.targetRemoteRepo}. Skip pushing.`);
```

Comment:
is this expected to happen?  Should it throw?

Reply:
I suppose this might happen, in case the pipeline run is canceled when branch is pushed but no PR is created

Reply:
Lets log this as an actual pipeline error
https://learn.microsoft.com/en-us/azure/devops/pipelines/scripts/logging-commands?view=azure-devops&tabs=powershell#logissue-log-an-error-or-warning